ssh_automated_tests/core/connection.py

```diff
@@ -19,12 +19,9 @@ def __init__(self, user: SshUser):
     def connect_via_key(self):
         print('connecting')
         connection_data = self.user.make_connection_data()
-        pkey = paramiko.RSAKey.from_private_key_file(connection_data["key"])
+        pkey = paramiko.RSAKey.from_private_key_file(connection_data.pop("key"))
         self.client.set_missing_host_key_policy(paramiko.AutoAddPolicy())
-        self.client.connect(hostname=connection_data["hostname"],
-                            username=connection_data["username"],
-                            port=connection_data["port"],
-                            pkey=pkey)
+        self.client.connect(**connection_data, pkey=pkey)
```

Comment:
IMHO, `a user` has to provide all required connection details with `user.paramiko_connection_data()` because using of `connection_data.pop("key")` or `connection_data["key"]` breaks encapsulation of connection details.